arcgis-ios-sdk-samples/Maps/Create and save a map/SaveAsViewController.swift

```diff
@@ -13,16 +13,23 @@
 // limitations under the License.
 
 import UIKit
+import ArcGIS
 
 protocol SaveAsViewControllerDelegate: AnyObject {
-    func saveAsViewController(_ saveAsViewController: SaveAsViewController, didInitiateSaveWithTitle title: String, tags: [String], itemDescription: String)
+    func saveAsViewController(_ saveAsViewController: SaveAsViewController, didInitiateSaveWithTitle title: String, tags: [String], itemDescription: String, folder: AGSPortalFolder?)
 }
 
 class SaveAsViewController: UITableViewController {
     @IBOutlet private weak var titleTextField: UITextField!
     @IBOutlet private weak var tagsTextField: UITextField!
     @IBOutlet private weak var descriptionTextField: UITextField!
+    @IBOutlet private weak var folderLabel: UILabel!
     
+    /// The array of folders loaded from the portal.
+    var portalFolders = [AGSPortalFolder]()
+    /// Folder selected by the user.
+    var selectedFolder: AGSPortalFolder?
```

Comment:
Instead of storing the selected folder, store the index of the selected folder. Then make this a computed property:

```swift
var selectedFolderIndex: Int?
var selectedFolder: AGSPortalFolder? { selectedFolderIndex.map { portalFolders[$0] } }
```